cmd/exo/cmd/request.go

```diff
@@ -0,0 +1,86 @@
+package cmd
+
+import (
+	"math/rand"
+	"sync"
+	"time"
+
+	"github.com/exoscale/egoscale"
+	"github.com/vbauerster/mpb"
+	"github.com/vbauerster/mpb/decor"
+)
+
+// asyncTask message variable must have same size with cmds
+func asyncTask(cmds []egoscale.AsyncCommand, message []string) []error {
+	errors := []error{}
+	taskBars := map[int]*mpb.Bar{}
```

Comment:
`taskBars := make(map[int]*mpb.Bar, len(cmds))`

Reply:
or `[len(cmds)]*mbp.Bar{}`

Reply:
`asyncTask` ~> `asyncTasks` as you accept more than one

Reply:
and it would probably be better to create a compound type

```golang
var flow = []task{
  {"deploy vm1", &deployVirtualMachine{...}},
  ...
}

_, errs := asyncTasks(flow...)
```

doing anything more complex than that means reinventing Terraform...